### PR TITLE
Add integration test suite and development documentation

### PR DESCRIPTION
**Pull request by lukehinds**
Introduce comprehensive shell-based integration tests that build nono and verify sandbox behavior from the user's perspective. Tests cover:

- Filesystem access control (read/write permissions, single file grants)
- Sensitive path protection (~/.ssh, ~/.aws, ~/.gnupg, etc.)
- System path protection (/usr/bin, /etc, /System/Library)
- Binary execution (shells, text tools, language runtimes)
- Network blocking (--net-block flag)
- Dangerous command blocking (rm, chmod, sudo, pip, etc.)
- Edge cases (symlinks, path variations, env vars, dry-run mode)

Tests gracefully handle platform differences (macOS TMPDIR behavior, Homebrew-installed runtimes) with skip functionality.

Also adds Development section to documentation with contributor guide and detailed testing documentation.

New files:
- tests/lib/test_helpers.sh
- tests/run_integration_tests.sh
- tests/integration/test_*.sh (7 test suites)
- docs/development/index.mdx
- docs/development/testing.mdx

Modified:
- .github/workflows/ci.yml (add integration job)
- docs/docs.json (add Development navigation)

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+updates:
+  # 1. Update your GitHub Actions (e.g., checkout, setup-rust)
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+
+  # 2. Update your Rust Toolchain (rust-toolchain.toml)
+  - package-ecosystem: "rust-toolchain"
```
[the remark this answers is not in the document]
Comment (lukehinds) [reply .github/dependabot.yml:11]: https://github.blog/changelog/2025-08-19-dependabot-now-supports-rust-toolchain-updates/
